ufl/formoperators.py

```diff
@@ -132,7 +133,9 @@ def adjoint(form, reordered_arguments=None):
     """
     form = as_form(form)
     form = expand_derivatives(form)
-    return compute_form_adjoint(form, reordered_arguments)
+    if isinstance(form, Form):
+        return compute_form_adjoint(form, reordered_arguments)
+    return Adjoint(form)
```

Comment:
Very nice! I guess the same needs to apply for `Action`, right?

Reply:
yep! hadn't got to that yet